src/convention/coerce.jl

```diff
@@ -168,3 +168,36 @@ function _check_tight(v::CArr, T, tight)
     end
     return v
 end
+
+
+## Images
+F = Union{Float32, FixedPointNumbers.Normed{UInt8,8}}
+_2d{C} =AbstractArray{C} where C<:Union{ColorTypes.AbstractRGB, ColorTypes.Gray}
+
+# Single Image
+function coerce(y::Arr{<:F, 2}, T2::Type{_2d{ColorTypes.Gray}})
+    return ColorTypes.Gray.(y)
+end
+
+function coerce(y::Arr{<:F, 3}, T2::Type{_2d{ColorTypes.AbstractRGB}})
+    return ColorTypes.RGB.(y)
+end
+
+# Collection
+
+_3Dcollection = AbstractArray{<:F, 3}
+_4Dcollection = AbstractArray{<:F, 4}
+
+function coerce(y::_3Dcollection, T2::Type{AbstractVector{_2d{ColorTypes.Gray}}})
+    return [ColorTypes.Gray.(y[:,:,idx]) for idx=1:size(y,3)]
+end
+
+function coerce(y::_4Dcollection, T2::Type{AbstractVector{_2d{ColorTypes.Gray}}})
+    @assert size(y, 3) == 1
```

Comment:
Instead of `@assert` please use 

```julia
size(y, 3) == 1 || error("Multiple color channels encountered. "*
                      "Perhaps you want to use `coerce(image_collection, ColorImage)`.")
```